setup.py

```diff
@@ -139,56 +148,157 @@ def download_dynamic_library():
+                dynfiles = [c for c in tar.getmembers() if c.name.endswith(suffixes)]
 
             with tarfile.open(filename, "r|gz", errorlevel=2) as tar:
                 print("extracting files:", [c.name for c in dynfiles])
                 tar.extractall(members=dynfiles, path=HERE)
-                os.rename(os.path.join(HERE, ddwaf_archive_dir), arch_dir)
+                os.rename(os.path.join(HERE, archive_dir), arch_dir)
 
-            # Rename ddwaf.xxx to libddwaf.xxx so the filename is the same for every OS
-            original_file = os.path.join(arch_dir, "lib", "ddwaf" + SUFFIX)
-            if os.path.exists(original_file):
-                renamed_file = os.path.join(arch_dir, "lib", "libddwaf" + SUFFIX)
-                os.rename(original_file, renamed_file)
+            # Rename <name>.xxx to lib<name>.xxx so the filename is the same for every OS
+            for suffix in suffixes:
+                original_file = os.path.join(arch_dir, "lib", cls.name + suffix)
+                if os.path.exists(original_file):
+                    renamed_file = os.path.join(arch_dir, "lib", "lib" + cls.name + suffix)
+                    os.rename(original_file, renamed_file)
 
             os.remove(filename)
 
+
+class LibDDWaf_Download(Library_Download):
+    name = "ddwaf"
+    download_dir = LIBDDWAF_DOWNLOAD_DIR
+    version = LIBDDWAF_VERSION
+    url_root = "https://github.com/DataDog/libddwaf/releases/download"
+    available_releases = {
+        "Windows": ["win32", "x64"],
+        "Darwin": ["arm64", "x86_64"],
+        "Linux": ["aarch64", "x86_64"],
+    }
+
+    @classmethod
+    def get_package_name(cls, arch, os):
+        archive_dir = "lib%s-%s-%s-%s" % (cls.name, cls.version, os.lower(), arch)
+        return archive_dir
+
+    @classmethod
+    def get_suffixes(cls, os):
+        TRANSLATE_SUFFIX = {"Windows": (".dll",), "Darwin": (".dylib",), "Linux": (".so",)}
+        return TRANSLATE_SUFFIX[os]
+
+    @classmethod
+    def run(cls):
+        LibDDWaf_Download.download_artifacts()
+
+
+class LibDatadog_Download(Library_Download):
+    name = "datadog"
+    download_dir = LIBDATADOG_PROF_DOWNLOAD_DIR
+    version = LIBDATADOG_PROF_VERSION
+    url_root = "https://github.com/DataDog/libdatadog/releases/download"
+    expected_checksums = {
+        "Linux": {
+            "x86_64": "e9ee7172dd7b8f12ff8125e0ee699d01df7698604f64299c4094ae47629ccec1",
+        },
+    }
+    available_releases = {
+        "Windows": [],
+        "Darwin": [],
+        "Linux": ["x86_64"],
+    }
+
+    @classmethod
+    def get_package_name(cls, arch, os):
+        osnames = {
+            "Linux": "unknown-linux-gnu",
+        }
+        tar_osname = osnames[os]
+        archive_dir = "lib%s-%s-%s" % (cls.name, arch, tar_osname)
+        return archive_dir
+
+    @classmethod
+    def get_suffixes(cls, os):
+        TRANSLATE_SUFFIX = {"Windows": (), "Darwin": (), "Linux": (".a", ".h")}
+        return TRANSLATE_SUFFIX[os]
```

Comment:
nit: if we make `TRANSLATE_SUFFIX` a class attribute, then we could implement this logic in the base class, and let subclasses simply override the `TRANSLATE_SUFFIX` attribute?

Reply:
Good call, taken in [08737cc](https://github.com/DataDog/dd-trace-py/pull/5940/commits/08737ccc1c2490835fe89d4438b8602e2076be2c)